backend/clap/src/main/java/zpi/ppea/clap/logic/BusinessLogicService.java

```diff
@@ -0,0 +1,85 @@
+package zpi.ppea.clap.logic;
+
+import data_store.Criterion;
+import data_store.DataStoreGrpc;
+import data_store.PartialRating;
+import data_store.RatingsSubmissionRequest;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class BusinessLogicService {
+
+    @GrpcClient("dataStore")
+    DataStoreGrpc.DataStoreBlockingStub dataStoreBlockingStub;
+
+    public Double getRatingPointsByArea(Integer submissionId) {
```

Comment:
Naming again, this method returns rating points for the entire submission, so I think `ByArea` is kinda misleading. I'd go for something like `calculateSubmissionRating`.

Reply:
ok